src/ssa.rs

```diff
@@ -118,24 +118,24 @@ pub struct SSAStyle {
     /// If [SSAStyle::border_style] is `1`, then this specifies the width of the outline around the
     /// text (in pixels).
     /// Values may be `0`, `1`, `2`, `3` or `4`.
-    pub outline: i8,
+    pub outline: f32,
     /// If [SSAStyle::border_style] is `1`, then this specifies the depth of the drop shadow behind
     /// the text (in pixels). Values may be `0`, `1`, `2`, `3` or `4`. Drop shadow is always used in
     /// addition to an outline - SSA will force an outline of 1 pixel if no outline width is given.
-    pub shadow: i8,
+    pub shadow: f32,
     /// Sets how text is "justified" within the Left/Right onscreen margins, and also the vertical
     /// placing.
     pub alignment: Alignment,
     /// Defines the Left Margin in pixels.
-    pub margin_l: i32,
+    pub margin_l: f32,
     /// Defines the Right Margin in pixels.
-    pub margin_r: i32,
+    pub margin_r: f32,
     /// Defines the Vertical Left Margin in pixels.
-    pub margin_v: i32,
+    pub margin_v: f32,
     /// Specifies the font character set or encoding and on multilingual Windows installations it
     /// provides access to characters used in multiple than one language. It is usually 0 (zero)
     /// for English (Western, ANSI) Windows.
-    pub encoding: i32,
+    pub encoding: f32,
```

Comment:
Why did you changed the type?

Reply:
In reality those properties can be floats.

Reply:
![image](https://github.com/user-attachments/assets/5d60ba1f-01d3-4395-9486-696a0c56d417)


I need to update types according to .ssa specification.

Reply:
Okay I see. I've only used [this spec](http://www.tcax.org/docs/ass-specs.htm) as reference when rewriting the ass parsing and interpreted the "Values may be ..." part as "values must be ...".
I don't want to sound know-it-all, but the documentation of libass is not a ass/ssa specification, they have stated that right on the top. But I think most of subtitle software that utilized ass subtitles is built around libass so having compatibility to it is a good thing ig